AbstractClient.php

```diff
@@ -91,36 +103,41 @@ protected function performRequest(Endpoint\EndpointInterface $endpoint)
         return $body;
     }
 
+    protected function getUriFromEndpoint(EndpointInterface $endpoint): string
+    {
+        $uri = $endpoint->uri();
+        if ($this->prependPath !== null) {
+            return $this->prependPath . $uri;
+        }
+
+        return $uri;
+    }
+
     /**
      * @return array<mixed>
      */
-    protected function buildOptions(Endpoint\EndpointInterface $endpoint) : array
+    protected function buildOptions(Endpoint\EndpointInterface $endpoint): array
     {
+        $options = [];
         if ($this->optionBuilder) {
             $optionBuilder = $this->optionBuilder;
 
             $options = $optionBuilder($endpoint);
-
-            if ($options !== null) {
-                return $options;
-            }
         }
 
         $params = $endpoint->params();
         $body = $endpoint->body();
         $formData = $endpoint->formData();
 
-        $options = [];
-
-        if (! empty($params)) {
+        if (! empty($params) && ! empty($options['query'])) {
```

Comment:
Isn't this a bit strange?

Reply:
What exactly do you mean? I've refactored it a bit.

Reply:
Ok, I was a bit confused when I wrote this comment.